### PR TITLE
re-orging options functionality

### PR DESCRIPTION
**Pull request by ashleygwilliams**

### CODE REVIEW

File: src/options/project/mod.rs [hunk 1]
```diff
@@ -55,9 +57,10 @@ impl Rust {
 
 struct JavaScript {}
 impl JavaScript {
-    fn read(&self) -> Options {
-        let package_json = fs::read_to_string(PACKAGE_JSON).unwrap();
-        serde_json::from_str(&package_json).unwrap()
+    fn read(&self) -> Result<Options, OrandaError> {
+        let package_json = fs::read_to_string(PACKAGE_JSON)?;
+        let data: Options = serde_json::from_str(&package_json)?;
+        Ok(data)
```
Comment (ashleygwilliams) [inline src/options/project/mod.rs]: fun fact, if i don't separate it out like this, serde insists i write a deserializer for the error type

File: src/options/project/mod.rs [hunk 2]
```diff
@@ -14,14 +15,14 @@ pub struct Options {
 }
 
 impl Options {
-    pub fn load() -> Option<Options> {
+    pub fn load() -> Result<Option<Options>, OrandaError> {
```
Comment (ashleygwilliams) [inline src/options/project/mod.rs]: return types like this make me want to rename Options to Config - thoughts @SaraVieira ?

Comment (SaraVieira) [reply src/options/project/mod.rs]: You don't like optional options? Ahah

Comment (SaraVieira) [reply src/options/project/mod.rs]: Super up for changing, any name ideas?

Comment (ashleygwilliams) [reply src/options/project/mod.rs]: was thinking "Config" just so we don't end up with an Option of Options that contain Option<T> ?

Comment (SaraVieira) [reply src/options/project/mod.rs]: I super approve this change

File: src/config/project.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    fn detect() -> Option<Type> {
+        if Rust::config().exists() {
+            Some(Type::Rust(Rust {}))
+        } else if JavaScript::config().exists() {
+            Some(Type::JavaScript(JavaScript {}))
+        } else {
+            None
+        }
```
Comment (Gankra) [inline src/config/project.rs]: I assume long-term we'll want to detect if both are present and ask for clarification..?

**Review by Gankra — APPROVED** [review]

File: src/config/project.rs (hunk 3 above)
Comment (ashleygwilliams) [reply src/config/project.rs]: yeah 100%

**Comment by ashleygwilliams** [comment]
ok, this has been open forever and is blocking further dev. i propose that this get merged and i have added the additional config functionality as issues.

**Comment by ashleygwilliams** [comment]
issue was already filed https://github.com/axodotdev/oranda-rs/issues/11

**Comment by SaraVieira** [comment]
I approve this idea

**Pull request merged**